### PR TITLE
Lpitu tema2

### PR DESCRIPTION
**Pull request by Lys4nder**

### CODE REVIEW

File: product.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#include "store.hpp"
```
Comment (enachebogdan) [inline product.cpp:1]: did you mean to include product.hpp instead? don't think you need store class in here.

File: product.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+//Constructors
+Product::Product(string name_, double price_) : name(name_), price(price_) {}
+
+Product:://Copy constructor
+Product::Product(const Product& p) : name(p.name), price(p.price) 
+{ cout << "copy constructor called\n"; }
```
Comment (enachebogdan) [inline product.cpp:8]: even for one-liner functions, for readability is better to use {} on different lines and use indentation, like you used for the destructor for example. you can apply this for getters and setters as well.

File: product.hpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <iostream>
+#include <string>
+using namespace std;
+
+class Product {
+    string name;
+    double price;
+    public:
```
Comment (enachebogdan) [inline product.hpp:9]: for better visibility, usually we indent public/private/protected differently from the members under them. You can either remove spacing for public: or add another tab to everything under the category.

File: product.hpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#pragma once
+#include <iostream>
+#include <string>
+using namespace std;
```
Comment (enachebogdan) [inline product.hpp:4]: usually we recommend against the using keyword for the entire namespace, instead write std::cout or whatever you need.

File: xmlClass.hpp [hunk 5]
```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include <iostream>
+#include <string>
+#include "rapidXML/rapidxml.hpp"
```
Comment (enachebogdan) [inline xmlClass.hpp:4]: good thing that you used an external parcer, it wouldn't be a problem if you implemented your own solution anyway.

File: store.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            {
+                product.setName("lapte");
+                product.setPrice(i+i/0.33);
+                products.push_back(product);
+            }
+    }
+
+    void Store::printProducts()
+    {
+        for (int i=0; i<products.size(); i++)
```
Comment (enachebogdan) [inline store.cpp:73]: try using range-based for loop or iterators to parce your vector

**Review by enachebogdan — CHANGES_REQUESTED** [review]
Great job, code loos good, you implemented most of the required functionality. My comments are only for small issues, i'll have a second look once you've added the file parsing logic.

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,4 @@
+/build
+/.vscode
+/rapidXML
```
[the remark this answers is not in the document]
Comment (Lys4nder) [reply .gitignore:3]: sure! i will make sure to remove the folder from .gitignore. Is is mandatory to modify the CMake file as well, as there are only hpp's, not cpp's in the library folder?

**Comment by Lys4nder** [comment]
am dat push la versiunea completa

File: XMLObject.cpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
+#include "XMLObject.hpp"
+
+const std::vector<Product> XMLReader::read()
+{
+            std::ifstream file(fname);
+            std::string xmlString((std::istreambuf_iterator<char>(file)), std::istreambuf_iterator<char>());
+            
+            if (file)
+                std::cout << "file '" << fname << "' opened\n";
```
Comment (CristianIonutIonescu) [inline XMLObject.cpp:9]: if the file not found/cannot be opened why are we still trying to read from it?

File: CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 3.22)
+
+project(tema2)
+add_executable(tema2 main.cpp product.cpp store.cpp XMLObject.cpp)
+
+include_directories(/usr/include/rapidxml)
```
Comment (CristianIonutIonescu) [inline CMakeLists.txt:6]: might not be needed if
1. target_include_directories(point to correct interface)
2. you should include it after you checked for the presence of the library

**Review by CristianIonutIonescu — CHANGES_REQUESTED** [review]
Mentioning here 2 things that I saw in multiple files:
1. [Order or declaring ](https://google.github.io/styleguide/cppguide.html#Declaration_Order)
2. ALL if/else/for/while statements must have the body clearly defined between { }even if it's one line:
```
if(..) {
} else {
}
for(...) {
}
//etc
```

**Comment by Lys4nder** [comment]
done 😄

**Review by CristianIonutIonescu — APPROVED** [review]

**Review by enachebogdan — APPROVED** [review]

**Pull request merged**